get_all_manifests.sh

```diff
@@ -6,20 +6,20 @@ GITHUB_URL="https://github.com"
 # component: notebook, dsp, kserve, dashbaord, cf/ray/kueue/trainingoperator, trustyai, modelmesh, modelregistry.
 # in the format of "repo-org:repo-name:branch-name:source-folder:target-folder".
 declare -A COMPONENT_MANIFESTS=(
-    ["codeflare"]="opendatahub-io:codeflare-operator:main:config:codeflare"
-    ["ray"]="opendatahub-io:kuberay:dev:ray-operator/config:ray"
-    ["kueue"]="opendatahub-io:kueue:dev:config:kueue"
-    ["data-science-pipelines-operator"]="opendatahub-io:data-science-pipelines-operator:main:config:data-science-pipelines-operator"
-    ["odh-dashboard"]="opendatahub-io:odh-dashboard:main:manifests:dashboard"
-    ["kf-notebook-controller"]="opendatahub-io:kubeflow:v1.7-branch:components/notebook-controller/config:odh-notebook-controller/kf-notebook-controller"
-    ["odh-notebook-controller"]="opendatahub-io:kubeflow:v1.7-branch:components/odh-notebook-controller/config:odh-notebook-controller/odh-notebook-controller"
-    ["notebooks"]="opendatahub-io:notebooks:main:manifests:notebooks"
-    ["trustyai"]="trustyai-explainability:trustyai-service-operator:main:config:trustyai-service-operator"
-    ["model-mesh"]="opendatahub-io:modelmesh-serving:release-0.12.0-rc0:config:model-mesh"
-    ["odh-model-controller"]="opendatahub-io:odh-model-controller:release-0.12.0:config:odh-model-controller"
-    ["kserve"]="opendatahub-io:kserve:release-v0.12.1:config:kserve"
-    ["modelregistry"]="opendatahub-io:model-registry-operator:main:config:model-registry-operator"
-    ["trainingoperator"]="opendatahub-io:training-operator:dev:manifests:trainingoperator"
+    ["codeflare"]="opendatahub-io:codeflare-operator:v1.5.0-odh:config:codeflare"
+    ["ray"]="opendatahub-io:kuberay:v1.1.2-odh:ray-operator/config:ray"
+    ["kueue"]="opendatahub-io:kueue:v0.7.0-odh-test:config:kueue"
+    ["data-science-pipelines-operator"]="opendatahub-io:data-science-pipelines-operator:v2.4.x:config:data-science-pipelines-operator"
+    ["odh-dashboard"]="opendatahub-io:odh-dashboard:v2.25.1-odh-release:manifests:dashboard"
```

Comment:
i am still puzzled by this part:
why do we use branch instead of tag from components?
if we ask for 
```
#Release#
     - component_name1|release-branch-url|release-notes-url
```
where this release-notes-url is used? 
@AjayJagan

Reply:
it will be used while creating the gh release ie. https://github.com/opendatahub-io/opendatahub-operator/releases/tag/v2.15.0

Reply:
so the release-branch-url is just a duplicated info? apart from downloading manifests, which can be used with tag.

Reply:
the thing is I found that sometime the branch URL and tags have different values... for instance
https://github.com/opendatahub-io/opendatahub-community/issues/146#issuecomment-1985902143
If this will not be the case, we could reduce the URLs added to one.

Reply:
they are and should be different.
branch can be accidentally updated with new commits after components tag which one they would like to release. 
thats why should always use tag, not branch unless tag is not applicable. 

@ykaliuta what you do think?  ^

Reply:
@zdtsw @AjayJagan  I think components need the flexibility to update/ add commits into ODH. (For e.g dashboard previously stated they preferred branches instead of tags.) I think we should just use the urls provided by the components irrespective of it being the branch or tag, unless there are any other concerns

Reply:
so we could use something like
`- component-name|branch-or-tag-url|(maybe release notes URL here as there is no way to know that URL 🤔 )?`

Reply:
Also I have a question:
the get_all_manifests.sh script says `"repo-org:repo-name:branch-name:source-folder:target-folder"`
So if we provide a tag URL here will it work? 🤔

Reply:
> they are and should be different. branch can be accidentally updated with new commits after components tag which one they would like to release. thats why should always use tag, not branch unless tag is not applicable.
> 
> @ykaliuta what you do think? ^

I agree with @zdtsw. For tagged releases we should use tagged components, otherwise the branch can be updated and we have very different build for the same tags.

For "incubation" it's ok to have branch.

From git and script point of view they are both refs and work in get_all_manifests.sh even with --depth 1 --branch <ref name>. Just it creates detached HEAD and prints warning then:

```
Cloning repo data-science-pipelines-operator: opendatahub-io:data-science-pipelines-operator:v2.0.1:config:data-science-pipelines-operator
X11 forwarding request failed
Note: switching to '5915222ac8ea82a6ad5e9d99705fd4a6ae719f43'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false
```

Reply:
> they are and should be different. branch can be accidentally updated with new commits after components tag which one they would like to release. thats why should always use tag, not branch unless tag is not applicable.

I've been bitten by misaligned branches in the past and I am happy that tags are now in use.